### DISCUSSION TITLE
openssl@1.1 upgrade doesn't complete

### DISCUSSION THREAD

**Discussion by mmuftwork** [Homebrew/discussions#296] [Everyday usage]
For about a week now, attempts to upgrade `openssl@1.1` and anything that requires it such as `python@3.9` fails at the point of `make test`:

```
==> Installing dependencies for python@3.9: openssl@1.1
==> Installing python@3.9 dependency: openssl@1.1
==> perl ./Configure --prefix=/usr/local/Cellar/openssl@1.1/1.1.1i --openssldir=/usr/local/etc/openssl@1.1 no-ssl3 no-ss
==> make
==> make install MANDIR=/usr/local/Cellar/openssl@1.1/1.1.1i/share/man MANSUFFIX=ssl
==> make test
```
The terminal window shows it switching between `perl` and `openssl` and then freezes like this:

<img width="709" alt="terminalwindowmessage" src="https://user-images.githubusercontent.com/38887910/102278134-47f4c580-3ef7-11eb-9f48-0e47d244d2af.png">

I've left it running for 48 hours and it never got past that.

**Comment by SMillerDev** [comment]
Why are you building from source? It can take a long time so that's normal.

**Reply by carlocab** [reply]
They're running High Sierra; that's why: https://github.com/Homebrew/discussions/discussions/278

**Reply by mmuftwork** [reply]
Yes, we're running High Sierra 10.13.6. Is there an option to download a compiled version through homebrew under High Sierra?

**Reply by carlocab** [reply]
Not from homebrew-core, unfortunately. Is it impossible for you to upgrade to, say, Mojave?

**Reply by mmuftwork** [reply]
I use Photoshop CS5 extensively. That doesn't run under Mojave.
Was it still building after 48 hours or had it frozen? I can try and leave it running for longer if I know it wasn't dead.

**Reply by carlocab** [reply]
I can't say for sure without having a look, but it's certainly possible. One way to try and check is to see if the associated processes are using any of your processors. You can look at this with, say, `top` (or maybe even the Activity Monitor app).

**Reply by SMillerDev** [reply]
I guess CS5 isn't supported on Mojave and Homebrew isn't supported on High Sierra so you'll have to pick one.

**Reply by gromgit** [reply]
Have a look in `~/Library/Logs/Homebrew/openssl@1.1`. You should see the build logs in there, and the timestamps on the various files should tell you whether it's "gone fishing" or still churning.

**Reply by mmuftwork** [reply]
I started the process again. It reached `make test`. By that point, there was `04.make` in the directory. I tailed it and it seems stuck here:
```
../test/recipes/70-test_comp.t ..................... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 4/4 subtests 
../test/recipes/70-test_key_share.t ................ 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
../test/recipes/70-test_packet.t ................... ok
../test/recipes/70-test_recordlen.t ................ ok
../test/recipes/70-test_renegotiation.t ............ ok
../test/recipes/70-test_servername.t ............... ok
```

**Reply by mmuftwork** [reply]
I'll let it keep running and update if there is any change.

**Reply by thepolygonist** [reply]
have you found a solution? i'm facing the same issue

**Reply by mmuftwork** [reply]
Over an hour and no changes to `04.make`. No indications of changes from the log. Seems stuck on whatever follows `../test/recipes/70-test_servername.t ............... ok`

**Reply by carlocab** [reply]
You really should check if the associated processes are consuming any resources. That's a much more reliable way of figuring out if something is happening or not.

**Reply by mmuftwork** [reply]
Ran `top`

```
ID    COMMAND      %CPU TIME     #TH   #WQ  #PORTS  MEM    PURG   CMPRS  PGRP  PPID  STATE    BOOSTS           %CPU_ME
98543  MTLCompilerS 0.0  00:00.06 2     2    22      8192B  0B     3036K  98543 1     sleeping  0[1]            0.00000
```

**Reply by mmuftwork** [reply]
`ps -ef | grep -i brew` showed:


  501  2059  2058   0  9:28AM ttys005    0:01.30 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-runtime-stub-0.2.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-macho-2.5.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-sorbet-0.5.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rspec-2.0.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rails-2.9.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-performance-1.9.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-progressbar-1.10.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rack-2.2.3/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/plist-3.5.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/patchelf-1.3.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mechanize-2.7.6/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/elftools-1.1.3/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov_json_formatter-0.1.2/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/bindata-2.4.8/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/activesupport-6.1.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/zeitwerk-2.4.2/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tzinfo-2.0.3/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/i18n-1.8.5/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/concurrent-ruby-1.1.7/lib/concurrent-ruby:/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0:/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/x86_64-darwin14:/usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb

**Reply by thepolygonist** [reply]
i tried again and somehow it worked, no idea why

**Reply by carlocab** [reply]
Ah well. Compilers can be fickle sometimes. Glad it's finally working.

**Reply by mmuftwork** [reply]
`brew reinstall openssl@1.1` again got to `make test`. Seems stuck on:

```
../test/recipes/70-test_servername.t ............... ok
../test/recipes/70-test_sslcbcpadding.t ............ ok
../test/recipes/70-test_sslcertstatus.t ............ ok
../test/recipes/70-test_sslextension.t ............. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 6/8 subtests 
```

**Reply by carlocab** [reply]
I think this might mean that it's no longer compatible with High Sierra, I'm afraid.

**Reply by mmuftwork** [reply]
Is there a way to brew downgrade `python` and `openssl` to versions that do work?

**Reply by SMillerDev** [reply]
Short answer: No because it would make your system an insecure unmaintainable mess.

Long answer: You could take over maintenance of these formula for yourself. You could use `brew extract` to migrate them to your own tap and make copies of all formulae that depend on it to point to the older version. You'd still have to compile them from source and it'll be a lot of work making copies, but at least it won't try to upgrade anymore (which is a bad thing for security tools like OpenSSL)

**Reply by gromgit** [reply]
And once you create your own tap, you should also consider taking advantage of the recently-introduced ability to upload bottles to GitHub Releases, so you only have to build it once for distribution to any other High Sierra boxes: https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/
